packages/compiler/src/server/compile-service.ts

```diff
@@ -94,6 +96,15 @@ export function createCompileService({
     updated.scheduleUpdate(document);
   }
 
+  /**
+   * Compile the given document.
+   * First, the main.tsp file will be obtained for compilation.
+   * If the current document is not the main.tsp file,
```

Comment:
it will also re-compile if the given doc is not included in the compilation starting from the main file found.